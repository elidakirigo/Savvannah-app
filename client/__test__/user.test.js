import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import Page from '../src/app/user/[id]/page'
import Navbar from '../components/navbar'
import '@testing-library/jest-dom'
import { render, fireEvent, screen, waitFor } from '@testing-library/react'
import StoreProvider from '../components/StoreProvider'
import * as nextAuthReact from 'next-auth/react'
import '@testing-library/jest-dom'

jest.mock('next-auth/react')

global.fetch = jest.fn()

describe('render the user page', () => {
	it('renders userpage unchanged', () => {
		const { container } = render(<Page />)
		expect(container).toMatchSnapshot()
	})

	it('expect api call to change ptag', async () => {
		const fakeUserResponse = { data: 'response' }
		var { getByTestId } = render(<Page />)
		var apiFunc = jest.spyOn(global, 'usefetchuser').mockImplementationOnce(() => {
			console.log('the response ' + JSON.stringify(response))
			return Promise.resolve({
				json: () => Promise.resolve(fakeUserResponse),
			})
		})

		fireEvent.click(getByTestId('apiCall'))
		const text = await getByTestId('ptag')
		expect(text).toHaveTextContent(fakeUserResponse['data'])
	})
})
