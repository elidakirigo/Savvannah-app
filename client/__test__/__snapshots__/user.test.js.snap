// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the user page should render userpage successfully 1`] = `
<div>
  <div
    class="relative overflow-x-auto shadow-md sm:rounded-lg max-w-[1000px] m-5"
  >
    <h1
      class="text-[30px]"
      data-testid="username"
    >
      name : 
      Leanne Graham
    </h1>
    <table
      class="w-full text-sm text-left rtl:text-right text-gray-500 dark:text-gray-400 mt-8"
    >
      <thead
        class="text-xs text-gray-700 uppercase dark:text-gray-400"
      >
        <tr>
          <th
            class="px-6 py-3 bg-gray-50 dark:bg-gray-800"
            data-testid="albumtitle"
            scope="col"
          >
            Album title
          </th>
          <th
            class="px-6 py-3"
            data-testid="albumid"
            scope="col"
          >
            ID
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="border-b border-gray-200 dark:border-gray-700 cursor-pointer "
        >
          <th
            class="px-6 py-4 font-medium text-gray-900 whitespace-nowrap bg-gray-50 hover:bg-slate-500 dark:text-white dark:bg-gray-800"
            data-testid="username"
            scope="row"
          >
            quidem molestiae enim
          </th>
          <td
            class="px-6 py-4"
          >
            1
          </td>
        </tr>
        <tr
          class="border-b border-gray-200 dark:border-gray-700 cursor-pointer "
        >
          <th
            class="px-6 py-4 font-medium text-gray-900 whitespace-nowrap bg-gray-50 hover:bg-slate-500 dark:text-white dark:bg-gray-800"
            data-testid="username"
            scope="row"
          >
            sunt qui excepturi placeat culpa
          </th>
          <td
            class="px-6 py-4"
          >
            2
          </td>
        </tr>
        <tr
          class="border-b border-gray-200 dark:border-gray-700 cursor-pointer "
        >
          <th
            class="px-6 py-4 font-medium text-gray-900 whitespace-nowrap bg-gray-50 hover:bg-slate-500 dark:text-white dark:bg-gray-800"
            data-testid="username"
            scope="row"
          >
            omnis laborum odio
          </th>
          <td
            class="px-6 py-4"
          >
            3
          </td>
        </tr>
        <tr
          class="border-b border-gray-200 dark:border-gray-700 cursor-pointer "
        >
          <th
            class="px-6 py-4 font-medium text-gray-900 whitespace-nowrap bg-gray-50 hover:bg-slate-500 dark:text-white dark:bg-gray-800"
            data-testid="username"
            scope="row"
          >
            non esse culpa molestiae omnis sed optio
          </th>
          <td
            class="px-6 py-4"
          >
            4
          </td>
        </tr>
        <tr
          class="border-b border-gray-200 dark:border-gray-700 cursor-pointer "
        >
          <th
            class="px-6 py-4 font-medium text-gray-900 whitespace-nowrap bg-gray-50 hover:bg-slate-500 dark:text-white dark:bg-gray-800"
            data-testid="username"
            scope="row"
          >
            eaque aut omnis a
          </th>
          <td
            class="px-6 py-4"
          >
            5
          </td>
        </tr>
        <tr
          class="border-b border-gray-200 dark:border-gray-700 cursor-pointer "
        >
          <th
            class="px-6 py-4 font-medium text-gray-900 whitespace-nowrap bg-gray-50 hover:bg-slate-500 dark:text-white dark:bg-gray-800"
            data-testid="username"
            scope="row"
          >
            natus impedit quibusdam illo est
          </th>
          <td
            class="px-6 py-4"
          >
            6
          </td>
        </tr>
        <tr
          class="border-b border-gray-200 dark:border-gray-700 cursor-pointer "
        >
          <th
            class="px-6 py-4 font-medium text-gray-900 whitespace-nowrap bg-gray-50 hover:bg-slate-500 dark:text-white dark:bg-gray-800"
            data-testid="username"
            scope="row"
          >
            quibusdam autem aliquid et et quia
          </th>
          <td
            class="px-6 py-4"
          >
            7
          </td>
        </tr>
        <tr
          class="border-b border-gray-200 dark:border-gray-700 cursor-pointer "
        >
          <th
            class="px-6 py-4 font-medium text-gray-900 whitespace-nowrap bg-gray-50 hover:bg-slate-500 dark:text-white dark:bg-gray-800"
            data-testid="username"
            scope="row"
          >
            qui fuga est a eum
          </th>
          <td
            class="px-6 py-4"
          >
            8
          </td>
        </tr>
        <tr
          class="border-b border-gray-200 dark:border-gray-700 cursor-pointer "
        >
          <th
            class="px-6 py-4 font-medium text-gray-900 whitespace-nowrap bg-gray-50 hover:bg-slate-500 dark:text-white dark:bg-gray-800"
            data-testid="username"
            scope="row"
          >
            saepe unde necessitatibus rem
          </th>
          <td
            class="px-6 py-4"
          >
            9
          </td>
        </tr>
        <tr
          class="border-b border-gray-200 dark:border-gray-700 cursor-pointer "
        >
          <th
            class="px-6 py-4 font-medium text-gray-900 whitespace-nowrap bg-gray-50 hover:bg-slate-500 dark:text-white dark:bg-gray-800"
            data-testid="username"
            scope="row"
          >
            distinctio laborum qui
          </th>
          <td
            class="px-6 py-4"
          >
            10
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
